internal/pkg/nats/client.go

```diff
@@ -137,6 +137,16 @@ func (c *Client) Subscribe(topics []types.TopicChannel, messageErrors chan error
 			} else {
 				tc.Messages <- env
 			}
+
+			if c.config.ExactlyOnce {
+				// AckSync carries a performance penalty
+				// but is needed for subscribe side of ExactlyOnce
+				if ackErr := msg.AckSync(); err != nil {
+					messageErrors <- ackErr
+				}
+			} else {
+				msg.Ack()
+			}
```

Comment:
Need to check if msg has a non-empty reply before attempting, AckError should also be handled for both cases.

Reply:
So is there more to be done here for this PR?